### PR TITLE
fix(ci): archive the generated code

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: ci/release/prepare.sh [hunk 1]
```diff
@@ -6,3 +6,7 @@ set -euo pipefail
 # build artifacts
 buf build
 buf generate
+
+for dir in gen/proto/*; do
+  tar -czf "$(dirname "${dir}")"/"${dir//\//.}.tar.gz" "${dir}"
```
Comment (cpcloud) [inline ci/release/prepare.sh:11]: The resulting tarballs are named like `gen.proto.${language}.tar.gz`.

Comment (jacques-n) [reply ci/release/prepare.sh:11]: Which languages is this useful for? I assume go would just use buf. For java, really need a maven coordinate jar file, etc.

Comment (cpcloud) [reply ci/release/prepare.sh:11]: Ah, so should we just remove it then?

Comment (cpcloud) [reply ci/release/prepare.sh:11]: All languages should be able to use buf if there's a protoc plugin, that said I don't know much about Java package management.

Comment (jacques-n) [reply ci/release/prepare.sh:11]: Yeah, I think we should remove this step. Java has something similar to npm or pip which is a separate process we'll need to undertake. Ideally buf would do the work but it doesn't look like they have focused on that ecosystem.

Comment (cpcloud) [reply ci/release/prepare.sh:11]: Roger that, I'll remove this and keep the conventionalcommits changes. That style is a bit better IMO. Here's the recent ibis release with that style: https://github.com/ibis-project/ibis/releases/tag/3.0.0

**Review by jacques-n — APPROVED** [review]
Sounds good. LGTM

**Pull request merged**